### DISCUSSION TITLE
[Bug]: Appending field array item does not reflect the nested field array

### DISCUSSION THREAD

**Discussion by repl-ullas-ml** [react-hook-form/react-hook-form#6610] [Q&A]
### Version Number

7.15.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-nested-usefieldarray-forked-x6uf0

### Steps to reproduce

1. Click on `Add Order`
2. Notice on `Add Order`, for a field array `orders` and order was added  ``` { title: "birds", items: [{ title: "sparrow" }] }``` 
3. An attribute of `order`  ie `items` is again a field array
3. Even though `items` with `sparrow` is added, it does not reflect on the fields

### Expected behaviour

Adding the order with `items` should reflect on the fields

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
`<button type="button" onClick={() => append({})}>`

why the append function is an empty object? it's required to have a payload for what you want to append.

**Comment by repl-ullas-ml** [comment]
@bluebill1049 sure, but it isn't something i intend to showcase as bug anyways. Please note i am adding an `Order` not an `Item`. Please see the attached gif

An `order` with `items` which consists of `sparrow` isn't reflected upon adding a `order`


![mix-match-collectors](https://user-images.githubusercontent.com/11545334/134406930-a650ba1f-635b-4f7e-b369-aaa55b8fde89.gif)

**Reply by bluebill1049** [reply]
and you are using v6 syntax in v7? 
```
<input
  ref={register()}
  name={`orders[${orderIndex}].title`}
  defaultValue={order.title}
/>
```
could you double-check your example against the doc?

**Reply by repl-ullas-ml** [reply]
I see, you are referring to this section in the doc https://react-hook-form.com/migrate-v6-to-v7/

![Screen Shot 2021-09-24 at 11 40 49 AM](https://user-images.githubusercontent.com/11545334/134717920-640b7fa9-50e7-4f01-989c-b90b7980e56d.png)

I have updated the codesandbox to reflect this and i still see the issue
https://codesandbox.io/s/react-hook-form-nested-usefieldarray-forked-x6uf0?file=/src/App.js

**Reply by bluebill1049** [reply]
could you look at the example: https://github.com/react-hook-form/react-hook-form/blob/master/app/src/useFieldArrayNested.tsx see what's missing? (you can run that example as well, just clone the repo yarn start:app)
